src/rcc.rs

```diff
@@ -438,32 +438,33 @@ impl CFGR {
     fn get_sysclk(&self) -> (u32, cfgr::SW_A, Option<PllConfig>) {
         // If a sysclk is given, check if the PLL has to be used,
         // else select the system clock source, which is either HSI or HSE.
-        if let Some(sysclk) = self.sysclk {
-            if let Some(hseclk) = self.hse {
-                if sysclk == hseclk {
+        match (self.sysclk, self.hse) {
+            (Some(sysclk), Some(hse)) => {
+                if sysclk == hse {
```

Comment:
You could even put this in an if-guard on the match to remove one level of indentation

Reply:
Good idea!

Reply:
Nice, with the help of if-guard, I could simplify it quite a bit further!